include/zephyr/drivers/flash.h

```diff
@@ -583,6 +583,84 @@ static inline const struct flash_parameters *z_impl_flash_get_parameters(const s
 __syscall int flash_ex_op(const struct device *dev, uint16_t code,
 			  const uintptr_t in, void *out);
 
+/**
+ * @brief Check if two flash regions on the same device overlap.
+ *
+ * @param a_start offset of first region.
+ * @param a_size size of first region.
+ * @param b_start offset of second region.
+ * @param b_size size of second region.
+ *
+ * @retval 0 if the regions do not overlap.
+ * @retval 1 if the regions overlap.
+ * @retval -EINVAL if any value provided is negative.
+ * @retval -EOVERFLOW if either region overflows the address space.
+ */
+int flash_regions_overlap(off_t a_start, off_t a_size, off_t b_start, off_t b_size);
```

Comment:
Can we use "range" instead of "region"? It is just that region somehow reminds me of something datasheet may describe as having specific properties in a device, like write-once region, protected region.

Reply:
Sure